overlord/snapstate/handlers.go

```diff
@@ -213,6 +213,13 @@ func (m *SnapManager) doPrerequisites(t *state.Task, _ *tomb.Tomb) error {
 }
 
 func (m *SnapManager) installOneBaseOrRequired(st *state.State, snapName, channel string, onInFlight error, userID int) (*state.TaskSet, error) {
+	// the core snap provides everything we need for core16
```

Comment:
this reminds that we should check early  both in isInstalled and with a flag to Install whether in the base/core case the snap has the right type

Reply:
Do you want this to be done in this PR or in a followup?

Reply:
It's kind of an orthogonal issue, or a prereq,  maybe somebody else can work on it?